applications/SwimmingDEMApplication/custom_elements/qsvms_dem_coupled.cpp

```diff
@@ -0,0 +1,859 @@
+            for (unsigned int d = 0; d < Dim;++d)
+                F[LocalIndex] -= rData.Weight * rData.N[i] * (fluid_fraction_rate - mass_source);
+            LocalIndex += Dim + 1;
+        }
+
+    }
+
+template<class TElementData>
+void QSVMSDEMCoupled<TElementData>::AddProjectionToRHS(VectorType& RHS,
+                                                    const array_1d<double, 3 > & rAdvVel,
+                                                    TElementData& rData,
+                                                    const double TauOne,
+                                                    const double TauTwo,
+                                                    const double Weight,
+                                                    const double DeltaTime)
+    {
+        const unsigned int BlockSize = Dim + 1;
+        const double density = this->GetAtCoordinate(rData.Density, rData.N);
+        Vector AGradN;
+        this->ConvectionOperator(AGradN, rAdvVel, rData.DN_DX); // Get a * grad(Ni)
+        array_1d<double,Dim> MomProj(Dim,0.0);
+        MomProj = this->GetAtCoordinate(rData.MomentumProjection, rData.N);
+        const auto& fluid_fraction_gradient = rData.FluidFractionGradient;
+        double DivProj = 0.0;
+        DivProj = this->GetAtCoordinate(rData.MassProjection, rData.N);
+        const double fluid_fraction = this->GetAtCoordinate(rData.FluidFraction, rData.N);
+        unsigned int FirstRow = 0;
+
+        for (unsigned int i = 0; i < NumNodes; i++)
+        {
+            for (unsigned int d = 0; d < Dim; d++)
+            {
+
+                RHS[FirstRow+d] -= Weight * (density * AGradN[i] * TauOne * MomProj[d] + (fluid_fraction * rData.DN_DX(i,d) + fluid_fraction_gradient(i,d) * rData.N[i]) * TauTwo * DivProj); // TauOne * (a * Grad(v)) * MomProjection + TauTwo * Div(v) * MassProjection
+                RHS[FirstRow+Dim] -= Weight * rData.DN_DX(i,d) * TauOne * MomProj[d]; // TauOne * Grad(q) * MomProjection
+            }
+            FirstRow += BlockSize;
+        }
+    }
+
+template< class TElementData >
+void QSVMSDEMCoupled<TElementData>::CalculateTau(
+    TElementData& rData,
+    const array_1d<double,3> &Velocity,
+    double &tau_one,
+    double &tau_two)
+{
+    constexpr double c1 = 8.0;
+    constexpr double c2 = 2.0;
+
+    const double h = rData.ElementSize;
+    const double density = this->GetAtCoordinate(rData.Density,rData.N);
+    this->GetEffectiveViscosity(rData, rData.EffectiveViscosity);
+    const double viscosity = this->GetAtCoordinate(rData.EffectiveViscosity,rData.N);
+    double velocity_norm = Velocity[0]*Velocity[0];
+    for (unsigned int d = 1; d < Dim; d++)
+        velocity_norm += Velocity[d]*Velocity[d];
+    velocity_norm = std::sqrt(velocity_norm);
+
+    double inv_tau = c1 * viscosity / (h*h) + density * (rData.DynamicTau/rData.DeltaTime + c2 * velocity_norm / h);
+    tau_one = 1.0/inv_tau;
+    tau_two = viscosity + c2 * density * velocity_norm * h / c1;
+
+}
```

Comment:
As far as I see these are exactly the same stabilization constants that the standard QSVMS uses. Why you need to reimplement it?